cli/app.go

```diff
@@ -812,84 +908,27 @@ var app = &cli.App{
 										},
 									},
-									Action: DataAddToDatasetByIDs,
+									Action: createCommandWithT[dataAddToDatasetByIDsArgs](DataAddToDatasetByIDs),
 								},
 								{
 									Name:      "filter",
 									Usage:     "adds binary data from the specified filter to dataset",
 									UsageText: createUsageText("dataset data add filter", []string{datasetFlagDatasetID}, true),
-									Flags: []cli.Flag{
+									// CR erodkin: flyby here, use commonFilterFlags
+									Flags: append([]cli.Flag{
 										&cli.StringFlag{
 											Name:     datasetFlagDatasetID,
 											Usage:    "dataset ID to which data will be added",
 											Required: true,
 										},
-										&cli.StringSliceFlag{
-											Name:  dataFlagOrgIDs,
-											Usage: "orgs filter",
-										},
-										&cli.StringSliceFlag{
-											Name:  dataFlagLocationIDs,
-											Usage: "locations filter",
-										},
-										&AliasStringFlag{
-											cli.StringFlag{
-												Name:    generalFlagMachineID,
-												Aliases: []string{generalFlagAliasRobotID},
-												Usage:   "machine id filter",
-											},
-										},
-										&cli.StringFlag{
-											Name:  dataFlagPartID,
-											Usage: "part id filter",
-										},
-										&AliasStringFlag{
-											cli.StringFlag{
-												Name:    dataFlagMachineName,
-												Aliases: []string{dataFlagAliasRobotName},
-												Usage:   "machine name filter",
-											},
-										},
-										&cli.StringFlag{
-											Name:  dataFlagPartName,
-											Usage: "part name filter",
-										},
-										&cli.StringFlag{
-											Name:  dataFlagComponentType,
-											Usage: "component type filter",
-										},
-										&cli.StringFlag{
-											Name:  dataFlagComponentName,
-											Usage: "component name filter",
-										},
-										&cli.StringFlag{
-											Name:  dataFlagMethod,
-											Usage: "method filter",
-										},
-										&cli.StringSliceFlag{
-											Name:  dataFlagMimeTypes,
-											Usage: "mime types filter",
-										},
-										&cli.StringFlag{
-											Name:  dataFlagStart,
-											Usage: "ISO-8601 timestamp indicating the start of the interval filter",
-										},
-										&cli.StringFlag{
-											Name:  dataFlagEnd,
-											Usage: "ISO-8601 timestamp indicating the end of the interval filter",
-										},
 										&cli.StringSliceFlag{
 											Name: dataFlagTags,
 											Usage: "tags filter. " +
 												"accepts tagged for all tagged data, untagged for all untagged data, or a list of tags for all data matching any of the tags",
 										},
-										&cli.StringSliceFlag{
-											Name: dataFlagBboxLabels,
-											Usage: "bbox labels filter. " +
-												"accepts string labels corresponding to bounding boxes within images",
-										},
 									},
-									Action: DataAddToDatasetByFilter,
+										commonFilterFlags...),
```

Comment:
(flyby) we were using all the `commonFilterFlags` here but rearticulating them with identical descriptions. Removed, moved to appending the `commonFilterFlags`.